### PR TITLE
Move modules

### PR DESCRIPTION
**Pull request by MichaelPonti**
I made this a separate edit, it was a bit bigger than I originally thought.

I tried to highlight where XF differs from WPF/UNO and I think I found them all, but I have not used modules with XF before, and even my experience with WPF has been pretty basic.

There are a couple of other small updates around Prism 7 to 8, .NET versions, TOC etc.

### CODE REVIEW

File: docs/modules.md [hunk 1]
```diff
@@ -209,9 +209,9 @@ The modules that an application can load are defined in a module catalog. The Pr
 
 The module catalog is represented by a class that implements the ```IModuleCatalog``` interface. The module catalog class is created by the ```PrismApplication``` base class during application initialization. Prism provides different implementations of module catalog for you to choose from. You can also populate a module catalog from another data source by calling the ```AddModule``` method or by deriving from ```ModuleCatalog``` to create a module catalog with customized behavior.
 
-By default, the ```App``` class, derived from ```PrismApplication```, creates a ```ModuleCatalog``` in the ```CreateModuleCatalog``` method. Override this method to use different types of ```ModuleCatalog```.
+By default, the ```App``` class, derived from ```PrismApplication```, creates a ```ModuleCatalog``` in the ```CreateModuleCatalog``` method. In WPF and UNO, you can override this method to use different types of ```ModuleCatalog```.
 
-#### Registering Modules in Code
+#### Registering Modules in Code (all platforms)
```
Comment (brianlagunas) [inline docs/modules.md]: no need to add the "all platforms"

File: docs/modules.md [hunk 2]
```diff
@@ -252,7 +252,7 @@ ModuleCatalog.AddModule(new ModuleInfo()
 });
 ```
 
-#### Registering Modules Using a XAML File
+#### Registering Modules Using a XAML File (WPF and UNO)
```
Comment (brianlagunas) [inline docs/modules.md]: don't need to add "(WPF and UNO)"

File: docs/modules.md [hunk 3]
```diff
@@ -312,7 +312,7 @@ To specify on-demand loading of your module, add the ```startupLoaded``` attribu
 <Modularity:ModuleInfo Ref="file://ModuleE.dll" moduleName="ModuleE" moduleType="ModuleE.Module, ModuleE, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" startupLoaded="false" />
 ```
 
-#### Registering Modules Using a Configuration File
+#### Registering Modules Using a Configuration File (WPF and UNO)
```
Comment (brianlagunas) [inline docs/modules.md]: don't need to add "(WPF and UNO)"

File: docs/modules.md [hunk 4]
```diff
@@ -370,7 +370,7 @@ To specify on-demand loading using a configuration file, set the ```startupLoade
 <module assemblyFile="ModularityWithUnity.Desktop.ModuleE.dll" moduleType="ModularityWithUnity.Desktop.ModuleE, ModularityWithUnity.Desktop.ModuleE, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" moduleName="ModuleE" startupLoaded="false" />
 ```
 
-#### Discovering Modules in a Directory 
+#### Discovering Modules in a Directory (WPF and UNO)
```
Comment (brianlagunas) [inline docs/modules.md]: don't need to add "(WPF and UNO)"

**Review by brianlagunas — CHANGES_REQUESTED** [review]

**Comment by MichaelPonti** [comment]
Thanks @brianlagunas , cleaned this one up as well.

**Review by brianlagunas — APPROVED** [review]

**Comment by brianlagunas** [comment]
You freaking ROCK!

**Pull request merged**